src/services/codefixes/importFixes.ts

```diff
@@ -0,0 +1,474 @@
+                }
+            }
+
+            return confirmedActions;
+
+            function checkSymbolHasMeaning(symbol: Symbol, meaning: SemanticMeaning) {
+                const declarations = symbol.getDeclarations();
+                return declarations ? some(symbol.declarations, decl => !!(getMeaningFromDeclaration(decl) & meaning)) : false;
+            }
+
+            function getCodeActionForImport(moduleSymbol: Symbol, isDefault?: boolean): ImportCodeAction[] {
+                // Check to see if there are already imports being made from this source in the current file
+                const existingDeclarations: (ImportDeclaration | ImportEqualsDeclaration)[] = [];
+                for (const importModuleSpecifier of sourceFile.imports) {
+                    const importSymbol = checker.getSymbolAtLocation(importModuleSpecifier);
+                    if (importSymbol === moduleSymbol) {
+                        existingDeclarations.push(getImportDeclaration(importModuleSpecifier));
+                    }
+                }
+
+                if (existingDeclarations.length > 0) {
+                    // With an existing import statement, there are more than one actions the user can do.
+                    return getCodeActionsForExistingImport(existingDeclarations);
+                }
+                else {
+                    return [getCodeActionForNewImport()];
+                }
+
+                function getImportDeclaration(moduleSpecifier: LiteralExpression) {
+                    let node: Node = moduleSpecifier;
+                    while (node) {
+                        if (node.kind === SyntaxKind.ImportDeclaration) {
+                            return <ImportDeclaration>node;
+                        }
+                        if (node.kind === SyntaxKind.ImportEqualsDeclaration) {
+                            return <ImportEqualsDeclaration>node;
+                        }
+                        node = node.parent;
+                    }
+                    return undefined;
+                }
+
+                function getCodeActionsForExistingImport(declarations: (ImportDeclaration | ImportEqualsDeclaration)[]): ImportCodeAction[] {
+                    const actions: ImportCodeAction[] = [];
+
+                    // It is possible that multiple import statements with the same specifier exist in the file.
+                    // e.g.
+                    //
+                    //     import * as ns from "foo";
+                    //     import { member1, member2 } from "foo";
+                    //
+                    //     member3/**/ <-- cusor here
+                    //
+                    // in this case we should provie 2 actions:
+                    //     1. change "member3" to "ns.member3"
+                    //     2. add "member3" to the second import statement's import list
+                    // and it is up to the user to decide which one fits best.

```

Comment:
Say the user already has a `import * as ns` in his code, which means he intends to use it somewhere, then the first choice might be good. And as you previously suggested, the second option might solve more errors because it addresses all identifiers with the same name in this file. However, I feel there are really no objective criteria regarding which one is "always" better, just provide two choices might be better than enforcing our own understanding.